Neos.ContentRepositoryRegistry/Classes/Service/EventMigrationService.php

```diff
@@ -298,7 +298,9 @@ public function migrateMetaDataToWorkspaceName(\Closure $outputFn): void
 
             // ... and update the event
             // the payload is only used for rebasing where we override the workspace either way:
-            $eventMetaData['commandPayload']['workspaceName'] = 'missing:' . ($eventMetaData['commandPayload']['contentStreamId'] ?? '');
+            $eventMetaData['commandPayload']['workspaceName'] = WorkspaceName::fromString(
+                'missing-' . \mb_substr($eventMetaData['commandPayload']['contentStreamId'] ?? '', 0, WorkspaceName::MAX_LENGTH - 8)
+            )->value;
```

Comment:
By using transliterateFromString everything should be captured. So this should be enough:

```suggestion
            $eventMetaData['commandPayload']['workspaceName'] = WorkspaceName::transliterateFromString(
                'missing-' . ($eventMetaData['commandPayload']['contentStreamId'] ?? '')
            )->value;
```

Reply:
That's almost what I suggested above. But in your example it would always lead to the md5-version if I get it right

Reply:
The md5 is just a fallback, if the cutting and lowercasing doesn't help.

Reply:
https://github.com/neos/neos-development-collection/blob/9.0/Neos.ContentRepository.Core/Classes/SharedModel/Workspace/WorkspaceName.php#L90-L93

Reply:
Ah right. No need for truncating from the outside 🤘🏻

Reply:
fine by me